examples/TestDriver/src/pages/basics.js

```diff
@@ -14,63 +14,74 @@ import {
 import { Switch as NbSwitch } from 'native-base';
 import { connect } from 'react-redux';
 
-import Heap from '@heap/react-native-heap';
+import Heap, { HeapIgnore, HeapCaptureRestrictor } from '@heap/react-native-heap';
 import { incrementAction, decrementAction } from '../reduxElements';
 import { makeSentinelButton } from '../sentinelUtilities';
 
+const HeapIgnoredNbSwitch = Heap.withHeapIgnore(NbSwitch, {
+  ignoreInteraction: false,
+  ignoreAllProps: true,
+  ignoreTargetText: false,
+  ignoreInnerHierarchy: false,
+});
+
 class BasicsPage extends Component {
   render() {
     return (
       <View style={styles.container}>
-        <NbSwitch testID="nbSwitch" />
+        <HeapIgnoredNbSwitch testID="nbSwitch" />
         <Switch testID="switch" />
-        <Button
-          testID="track1"
-          title="Call Track1"
-          onPress={() => Heap.track('pressInTestEvent1', {})}
-        />
-        <Button
-          testID="track2"
-          title="Call Track2"
-          onPress={() => Heap.track('pressInTestEvent2', {})}
-        />
-        <Button
-          testID="track3"
-          title="Call Track3"
-          onPress={() => Heap.track('pressInTestEvent3', {})}
-        />
-        <Button
-          testID="track4"
-          title="Call Track4"
-          onPress={() => Heap.track('pressInTestEvent4', {})}
-        />
-        <Button
-          testID="aup"
-          title="Call AUP"
-          onPress={() => Heap.addUserProperties({ prop1: 'foo', prop2: 'bar' })}
-        />
-        <Button
-          testID="identify"
-          title="Call Identify"
-          onPress={() => Heap.identify('foobar')}
-        />
-        <Button
-          testID="aep"
-          title="Add Event Properties"
-          onPress={() =>
-            Heap.addEventProperties({ eventProp1: 'bar', eventProp2: 'foo' })
-          }
-        />
-        <Button
-          testID="removeProp"
-          title="Remove eventProp1"
-          onPress={() => Heap.removeEventProperty('eventProp1')}
-        />
-        <Button
-          testID="clearProps"
-          title="Clear Event Properties"
-          onPress={() => Heap.clearEventProperties()}
-        />
+        <HeapCaptureRestrictor ignoreTargetText={true}>
```

Comment:
High level - what's the semantic difference between `HeapCaptureRestrictor` and `HeapIgnore`?  My first thought looking at this is that I'm not sure why they don't do the same thing...

Reply:
`HeapIgnore` ignores _everything_ by default (i.e. you'd have to explicitly pass `ignoreInteraction={false}` if you want to capture _anything_).  I.e. it's best if you want to ignore some piece of the hierarchy.

`HeapCaptureRestrictor` captures everything by default (i.e. if you passed `ignoreTarget={true}` it would restrict capture to everything that isn't target text).  I.e. it's best if you want to restrict what we capture to some extent.

My thinking is that there would be cases where it's only "safe" to capture only the fact that there was any interaction with a component (if it's safe to capture anything), but there might also be cases where you just want to leave out target text and props for a few buttons.

Reply:
It's also worth noting that `HeapCaptureRestrictor` was merely in a comment on the original RFC, but once I started testing, I got a little frustrated about having to specify 3 different props when I really just wanted to ignore the target text of something.  I'll probably remove it before I merge this particular PR, just to keep this narrowed down to a subset of the feature scope in the RFC (although I'd be interested in revisiting the concept in the future)

Reply:
Could this be a method instead of a component, maybe?

The way I'm envisioning this, `Heap.withHeapIgnore` and `Heap.withRestrictedCapture` might set up `<HeapIgnore>` components with defaults and be the "API" for `<HeapIgnore>`, but users can use `<HeapIgnore>` directly for more granular control if they want to.